devito/profiler.py

```diff
@@ -167,69 +179,70 @@ def _get_for_flops(self, name, loop, loads):
-                a_flops = self._get_assign_flops(elem, loads)
+                a_flops = self._get_assign_flops(elem, loads, to_ignore)
                 block_flops += a_flops
                 block_oi += a_flops
             elif isinstance(elem, Block):
-                nblock_oi, nblock_flops = self._get_block_oi_and_flops(name, elem, loads)
+                nblock_oi, nblock_flops = self._get_block_oi_and_flops(
+                    name, elem, loads, to_ignore)
                 block_oi += nblock_oi
                 block_flops += nblock_flops
             elif isinstance(elem, For):
-                block_oi += self._get_for_flops(name, elem, loads)
+                block_oi += self._get_for_flops(name, elem, loads, to_ignore)
             else:
                 # no op
                 pass
 
         return block_oi, block_flops
 
-    def _get_assign_flops(self, assign, loads):
+    def _get_assign_flops(self, assign, loads, to_ignore):
         flops = 0
-        cur_load = ""
         loads["stores"] += 1
 
-        idx = 0
-        brackets = 0
         # removing casting statements and function calls to floor
         # that can confuse the parser
-        string = (assign.lvalue + " " + assign.rvalue)\
-            .replace("float", '').replace("int", '').replace("floor", '')
-
-        while idx < len(string):
-            char = string[idx]
-            if len(cur_load) == 0:
-                if char == '[':
-                    brackets += 1
-                elif char == ']':
-                    brackets -= 1
-                elif (char in "+-*/" and string[idx - 1] is not 'e' and not
-                      string[idx + 1].isdigit() and brackets == 0):
-                    flops += 1
-                elif (char.isalpha() and not
-                      string[idx - 1].isdigit() and char not in "it"):
-                    cur_load += char
-                idx += 1
-            else:
-                if char is '[':
-                    loads[cur_load] += 1
-                    cur_load = ""
-                    brackets += 1
-                    idx += 1
-                elif char is ' ' and brackets == 0 and len(cur_load) > 0:
-                    loads[cur_load] += 1
-                    cur_load = ""
-                    idx += 1
-                else:
-                    cur_load += char
-                    idx += 1
-
-        if len(cur_load) > 0:
-            loads[cur_load] += 1
+        string = assign.lvalue + " " + assign.rvalue
+
+        to_ignore = [
+            "int",
+            "float",
+            "double",
+            "F",
+            "i",
+            "t",
+            "fabsf",
+            "e",
+            "temp",
+            "p",  # This one shouldn't be here.
+                  # It should be passed in by an Iteration object.
+                  # Added only because tti_example uses it.
+        ] + to_ignore
+
+        symbols = re.findall(r"[a-z_]+\d?", string)
```

Comment:
short comment for people who don't know reg exprs or that have to re-learn reg exprs every time they see something slightly more advanced ?